### PR TITLE
[Action Refactor] (No.3) Remove dependency on Backgroundable/Promptable/Telemetrable on Function App

### PR DESCRIPTION
**Pull request by shenqianjin**
…le/Telemetrable on Function App

### CODE REVIEW

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java [hunk 1]
```diff
@@ -36,6 +38,7 @@
 
     private Runnable runnable;
     private AzureActionEnum action;
+    private String doingName;
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java:41]: Rename this parameter? like `actionName` ?

Comment (shenqianjin) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java:41]: It is not an Action Name, it is a action DOING name such as starting/stopping/etc.

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNode.java [hunk 2]
```diff
@@ -116,149 +123,42 @@ protected void refreshItems() {
...
-            return Node.getProgressMessage(AzureActionEnum.DELETE.getDoingName(), DeploymentSlotModule.MODULE_NAME, DeploymentSlotNode.this.name);
-        }
-
-        @Override
-        public TelemetryParameter getTelemetryParameter() {
-            return TelemetryParameter.WebApp.DeploymentSlot.DELETE;
-        }
+    @AzureOperation(value = "start deployment slot", type = AzureOperation.Type.ACTION)
+    private void start() {
+        AzureWebAppMvpModel.getInstance().startDeploymentSlot(subscriptionId, webAppId, slotName);
```
Comment (Flanker32) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNode.java:128]: In function we get the status after start/stop/restart but deployment slot just render the action directly, shall we unify the behaviors?

Comment (shenqianjin) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNode.java:128]: Will unity them including WebApp in a individual PR later.

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java (hunk 1 above)
Comment (Flanker32) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/BasicActionBuilder.java:41]: We may sync later for this parameter

**Review by Flanker32 — APPROVED** [review]

**Pull request merged**